src/msg/msg_types.h

```diff
@@ -543,12 +547,67 @@ WRITE_CLASS_ENCODER_FEATURES(entity_addr_t)
 
 std::ostream& operator<<(std::ostream& out, const entity_addr_t &addr);
 
-inline bool operator==(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) == 0; }
-inline bool operator!=(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) != 0; }
-inline bool operator<(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) < 0; }
-inline bool operator<=(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) <= 0; }
-inline bool operator>(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) > 0; }
-inline bool operator>=(const entity_addr_t& a, const entity_addr_t& b) { return memcmp(&a, &b, sizeof(a)) >= 0; }
+inline bool operator==(const entity_addr_t& a, const entity_addr_t& b) {
+  struct sockaddr_storage ssa = a.get_sockaddr_storage();
```

Comment:
Is this include used in 'C' files? if not - no need for the 'struct'

Reply:
> Is this include used in 'C' files? if not - no need for the 'struct'

Let me try to rebuild ceph with and without the struct keyword and find if it is included :-)

Reply:
Here was a discussion, therefore was found that all the struct info should be used for comparison/sorting
[https://github.com/ceph/ceph/pull/45095](https://github.com/ceph/ceph/pull/45095)

Reply:
> Is this include used in 'C' files? if not - no need for the 'struct'

The struct keyword can be removed.